### ISSUE TITLE
about eval.py and demo_inference.py different reslut 

### ISSUE THREAD

**Issue by ghost** [OmmmmooooO/Attention_OCR#1]
There are only numbers in my training samples.
When I run # python eval.py --split_name test --train_log_dir , it shows good results,
![webwxgetmsgimg](https://user-images.githubusercontent.com/33537489/33356640-85806fe2-d4f8-11e7-9363-265ddded3212.jpg)

accuracy get 0.99.

but wehn I predict one picture only , converted the image to [150,600,3],
python demo_inference.py –batch_size=1 –checkpoint=
--image_path_pattern=/home..../pic/%02d.png ,None of the predicted strings are right.
;;;;;;;
![webwxgetmsgimg 1](https://user-images.githubusercontent.com/33537489/33356645-8f6c666e-d4f8-11e7-99c6-cabe9728b118.jpg)

This a problem that puzzled me for a long time.
@OmmmmooooO  @alexgorban    can you tell me the reason,thank you very much.

**Comment by OmmmmooooO** [comment]
I don't have demo_inference.py in this project nor tried inference.
So I'm  probably not able to answer this problem.

If you are talking about this:
https://github.com/tensorflow/models/pull/2423/commits/00d0d4b0e5f235b68041d47daaddf2dd0469f511#diff-d89ed9916e9120c64923a410c885876f

Trying to be helpful:

1. demo_inference.py only fits FSNS-like image as infernece.
Is fsns_train_%02d.png similar to fsns dataset?

2. FSNS dataset null_code 133 is u'\u2591' or 'LIGHT SHADE'. I'm not sure what it will be
in terminal commnad line but it looks exactly 'LIGHT SHADE' in tensorboard.

3. Did you forget assigning a pre-trained model? –checkpoint= ? 

4. get_dataset_image_size() looks odd.

**Comment by ghost** [comment]
@OmmmmooooO     thank you for your guidance
demo_inference.py
[](url) https://github.com/tensorflow/models/blob/master/research/attention_ocr/python/demo_inference.py


I use your program to generate datasets, and my training and testing samples are 16-digit numbers.It looks all right. In the training process, the loss function reaches 30。

python demo_inference.py –batch_size=1 --checkpoint= --image_path_pattern=
Should I add this item:  --checkpoint_inception=./inception_v3   ?
add it , the result is still not good

**Comment by OmmmmooooO** [comment]
It looks you use customized dataset to train, test and evaluate correctly and have a good result.
But works awful while inferencing?

If so, I might check charset in demo_inference.py.

**Comment by ghost** [comment]
@OmmmmooooO   
I use  my own dataset.50,000 samples, only numbers(16 Arabic numerals), 10 hours of training, loss  30.

If the contents of the picture is 9788, its label is 9788(name is 9788.png), do not add unk , right?

**Comment by alexgorban** [comment]
Multiple users reported this issue after [#2776](https://github.com/tensorflow/models/pull/2776) was merged. I will take a look on this issue.

**Comment by ghost** [comment]
@alexgorban  
Does this mean that the result of  eval.py  is correct and   the demo_inference.py  is wrong ?
Thank you very much teacher.

**Comment by ghost** [comment]
@alexgorban   dear teacher:
        If my samples label is Chinese(non ascii string ),unicode string , I will encounter an error, can't use tf.train.Feature(bytes_list=tf.train.BytesList(value=value)) generate data.Can you give me some hints, where do I need to make changes?
I make image/text=''  ， Will it affect the result by doing so?

**Comment by alexgorban** [comment]
I'm still investigating the core problem.

In order to use the model with your own data you will need to prepare a character set (see [charset_size=134.txt](https://download.tensorflow.org/data/fsns-20160927/charset_size=134.txt)) and store encoded stings as vector of integers into 'image/class' and 'image/unpadded_class' features. The 'image/text' feature is not used by the code. For future reference: this question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by ghost** [comment]
@alexgorban   I got it, thanks for your reminder. Thank you，  teacher.

**Comment by OmmmmooooO** [comment]
@ccnankai 
You can post your question o StackOverflow and leave the url here since it is not a bug.
I will close this issue after you leave the clue.

**Comment by ghost** [comment]
@OmmmmooooO   I got it. Thank you very much.
